test/Options/SimpleOption.js

```diff
@@ -1,21 +1,358 @@
+    // Value of the contract is notional * (underlying price - strike price)
+    const currentUnderlyingPrice = 800;
+    await updateEquityPrice(currentUnderlyingPrice);
+    var simpleOption = await getSimpleOption();
+
+    expect(await simpleOption.connect(option_buyer).valuation()).to.equal(0);
+  });
+
+  it("Settlement amount changed when underlying price changed", async function () {
+    // Settlement amount of the contract is value * underlying price
+    const currentUnderlyingPrice = 1200;
+    await updateEquityPrice(currentUnderlyingPrice);
+    var simpleOption = await getSimpleOption();
+
+    expect(await simpleOption.connect(option_seller).valuation()).to.equal(
+      20000
+    );
+    expect(
+      await simpleOption.connect(option_seller).settlementAmount()
+    ).to.equal(24000000);
+  });
+
+  it("Balance of seller account reduced after exercise", async function () {
+    var sellerBalanceBeforeExercise = await settlementToken.balanceOf(
+      option_seller.address
+    );
+
+    const currentUnderlyingPrice = 1200;
+    await updateEquityPrice(currentUnderlyingPrice);
+    var simpleOption = await getSimpleOption();
+
+    var settlementAmount = await simpleOption
+      .connect(option_seller)
+      .settlementAmount();
+    await settlementToken
+      .connect(option_seller)
+      .approve(simpleOption.contractAddress(), settlementAmount);
+    console.log("Settlement amount for SimpleOption is " + settlementAmount);
+
+    expect(await simpleOption.connect(option_buyer).exercise()).to.emit(
+      simpleOption,
+      "Exercised"
+    );
+    // Settlement amount of the contract is value * underlying price(24000000)
+    expect(await settlementToken.balanceOf(option_seller.address)).to.equal(
+      Number(sellerBalanceBeforeExercise) - Number(settlementAmount)
+    );
+  });
+
+  it("Balance of buyer account increased after exercise", async function () {
+    var buyerBalanceBeforeExercise = await settlementToken.balanceOf(
+      option_buyer.address
+    );
+
+    const currentUnderlyingPrice = 1200;
+    await updateEquityPrice(currentUnderlyingPrice);
+    var simpleOption = await getSimpleOption();
+
+    var settlementAmount = await simpleOption
+      .connect(option_seller)
+      .settlementAmount();
+    await settlementToken
+      .connect(option_seller)
+      .approve(simpleOption.contractAddress(), settlementAmount);
+    console.log("Settlement amount for SimpleOption is " + settlementAmount);
+
+    expect(await simpleOption.connect(option_buyer).exercise()).to.emit(
+      simpleOption,
+      "Exercised"
+    );
+    // Settlement amount of the contract is value * underlying price(24000000)
+    expect(await settlementToken.balanceOf(option_buyer.address)).to.equal(
+      Number(buyerBalanceBeforeExercise) + Number(settlementAmount)
+    );
+  });
+
+  // Setup a contract based on premium in USD, settile in CNY
+  beforeEach(async function () {
```

Comment:
Overall this is complicated, I think as we progress we should break things down into sub JS files and pull in via requires("") - this has been done in scripts\deploy\deploy.js - which is the first time it became aparent to me how complex things are getting. I'll also go back and refactor test\Integration. 

This will also mean that there will be a library of these js utilities .. which at the moment just sit under scripts\deploy - but have wider use such as here.

I propose this is OK for now and we go back in a separate exercise and do that refactor.